ghost/members-importer/lib/MembersCSVImporter.js

```diff
@@ -33,24 +34,27 @@ module.exports = class MembersCSVImporter {
      * @param {Function} options.getTimezone - function returning currently configured timezone
      * @param {() => Object} options.getMembersRepository - member model access instance for data access and manipulation
      * @param {() => Promise<import('@tryghost/tiers/lib/Tier')>} options.getDefaultTier - async function returning default Member Tier
+     * @param {() => Promise<import('@tryghost/tiers/lib/Tier')>} options.getTierByName - async function returning Member Tier by name
      * @param {Function} options.sendEmail - function sending an email
      * @param {(string) => boolean} options.isSet - Method checking if specific feature is enabled
      * @param {({job, offloaded, name}) => void} options.addJob - Method registering an async job
      * @param {Object} options.knex - An instance of the Ghost Database connection
      * @param {Function} options.urlFor - function generating urls
      * @param {Object} options.context
      */
-    constructor({storagePath, getTimezone, getMembersRepository, getDefaultTier, sendEmail, isSet, addJob, knex, urlFor, context}) {
+    constructor({storagePath, getTimezone, getMembersRepository, getDefaultTier, getTierByName, sendEmail, isSet, addJob, knex, urlFor, context}) {
         this._storagePath = storagePath;
         this._getTimezone = getTimezone;
         this._getMembersRepository = getMembersRepository;
         this._getDefaultTier = getDefaultTier;
+        this._getTierByName = getTierByName;
         this._sendEmail = sendEmail;
         this._isSet = isSet;
         this._addJob = addJob;
         this._knex = knex;
         this._urlFor = urlFor;
         this._context = context;
+        this._tierCache = new Map();
```

Comment:
I can't see that we ever clear this cache - which means we could be dealing with outdated values.

Reply:
[It gets cleared on each execution ](https://github.com/TryGhost/Ghost/pull/17506/files#diff-a34d8cdc0a7caff73fbddfd0f4e205fb9b247d1b4e1b8b648c05f40a29812a79R121-R122) 🙂